Controls/Cafe/Customer1.c

```diff
@@ -19,11 +25,29 @@ void Customer1_checkout(char** name, int* cost) {
  */
 void Task_Customer_1(void* p_arg) {
 	OS_ERR err;	// Make sure to check for errors and print the error code if not OS_ERR_NONE
-	
+	CPU_TS ts; 
 	// Use these values as parameters for Customer1_checkout
 	char* name;
 	int cost;
+	CPU_INT16U seconds = 4;//4000ms = 4s DELAY
+	OS_OPT opt =OS_OPT_TIME_HMSM_STRICT;
+	
 	while(1){
-		
+		OSMutexPend(&RegisterOccupied_Mutex, 0, OS_OPT_PEND_BLOCKING, &ts, &err);//pendblocking , pendnonblocking doesnt wait, just goes through codes
+		if(err!=OS_ERR_NONE){
+			printf("Error from OSMutexPend Customer 1");
+			return;
+		}
+		Customer1_checkout(&name,&cost);
+		OSMutexPost(&RegisterOccupied_Mutex, OS_OPT_POST_NONE, &err);
+		if(err!=OS_ERR_NONE){
+			printf("Error from OSMutexPost Customer 1");
+			return;
+		}
+		OSTimeDlyHMSM(0,0,seconds,0, opt,&err);
+		if(err!=OS_ERR_NONE){
+			printf("Error from OSTimeDlyHMSM Customer 1");
```

Comment:
You can print the error code from err itself as a hexadecimal number in this printf statement.